src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/RestoreAzureRMSqlDatabase.cs

```diff
@@ -147,6 +233,45 @@ public class RestoreAzureRmSqlDatabase
         [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
         public SwitchParameter AsJob { get; set; }
 
+        /// <summary>
+        /// Gets or sets the compute generation of the database copy
+        /// </summary>
+        [Parameter(ParameterSetName = FromPointInTimeBackupWithVcoreSetName, Mandatory = true,
+            HelpMessage = "The compute generation of the restored Azure Sql Database.")]
+        [Parameter(ParameterSetName = FromDeletedDatabaseBackupWithVcoreSetName, Mandatory = true,
+            HelpMessage = "The compute generation of the restored Azure Sql Database.")]
+        [Parameter(ParameterSetName = FromGeoBackupWithVcoreSetName, Mandatory = true,
+            HelpMessage = "The compute generation of the restored Azure Sql Database.")]
+        [Parameter(ParameterSetName = FromLongTermRetentionBackupWithVcoreSetName, Mandatory = true,
+            HelpMessage = "The compute generation of the restored Azure Sql Database.")]
+        [Alias("Family")]
+        [PSArgumentCompleter("Gen4", "Gen5")]
+        [ValidateNotNullOrEmpty]
+        public string ComputeGeneration { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Vcore numbers of the database copy
+        /// </summary>
+        [Parameter(ParameterSetName = FromPointInTimeBackupWithVcoreSetName, Mandatory = true,
+            HelpMessage = "The Vcore numbers of the restored Azure Sql Database.")]
+        [Parameter(ParameterSetName = FromDeletedDatabaseBackupWithVcoreSetName, Mandatory = true,
+            HelpMessage = "The Vcore numbers of the restored Azure Sql Database.")]
+        [Parameter(ParameterSetName = FromGeoBackupWithVcoreSetName, Mandatory = true,
+            HelpMessage = "The Vcore numbers of the restored Azure Sql Database.")]
+        [Parameter(ParameterSetName = FromLongTermRetentionBackupWithVcoreSetName, Mandatory = true,
+            HelpMessage = "The Vcore numbers of the restored Azure Sql Database.")]
+        [Alias("Capacity")]
+        [ValidateNotNullOrEmpty]
+        public int Vcore { get; set; }
```

Comment:
I'm sorry to go back and forth on this. In design meeting we agreed to call it `VCore` with big 'C. Sorry, can you change it (in all cmdlets)?